x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/IntBlockHash.java

```diff
@@ -59,20 +59,20 @@ public void add(Page page, GroupingAggregatorFunction.AddInput addInput) {
                     addInput.add(0, groupIds);
                 }
             } else {
-                try (IntVector groupIds = add(intVector)) {
-                    addInput.add(0, groupIds);
+                try (IntBlock groupIds = add(intVector)) {
+                    addInput.add(0, groupIds.asVector());
```

Comment:
asVector() here should return a non-null value here. However, this change is unsafe, as asVector() is allowed to return null. I think we should inline this or use `finally` to close the Block.